### DISCUSSION TITLE
"The resource you are attempting to access is locked" Error During Folder Rename

### DISCUSSION THREAD

**Discussion by adamfisher** [pnp/pnpcore#1438] [Q&A]
I am not seeing much guidance on this error I get when attempted to rename different folders in SharePoint. I verified the folder I am trying to rename had no shared links and nothing check out to a user (from what I could tell). I was able to successfully go in through the UI and do a rename though.

Here is the rename code. I also added a call right before to delete the shared links because I heard that can also cause this error during a rename but the folder I tested with didn't have any shared links so added it just in case:

```csharp
IFolder folder = GetMyFolder();
await folder.DeleteShareLinksAsync();
await folder.RenameAsync(newFolderName);
await folder.LoadAsync(p => p.Name, p => p.ServerRelativeUrl);
```

Gives me this error sometimes (but not always) on certain folders:
```
{HttpResponseCode: 423
Code: notAllowed
Message: The resource you are attempting to access is locked
ClientRequestId: ad9e927b-7920-4f25-b5a7-fde1282d6535
code: resourceLocked
date: 2024-04-08T13:50:41
client-request-id: ad9e927b-7920-4f25-b5a7-fde1282d6535
}
```

**Stacktrace:**
```
   at PnP.Core.Services.BatchClient.<ExecuteMicrosoftGraphInteractiveAsync>d__48.MoveNext()
   at PnP.Core.Services.BatchClient.<ExecuteMicrosoftGraphBatchAsync>d__39.MoveNext()
   at PnP.Core.Services.BatchClient.<ExecuteBatch>d__37.MoveNext()
   at PnP.Core.Model.BaseDataModel`1.<RequestAsync>d__96.MoveNext()
   at PnP.Core.Model.SharePoint.Folder.<RenameAsync>d__74.MoveNext()
   at ....
```

**Comment by jansenbe** [comment]
@adamfisher : is the same folder sometimes failing and sometimes working? Did you check that the site is not read-only?

**Reply by adamfisher** [reply]
The site has many different folders and we successfully renamed other folders. There was just something special about this one for some reason but I couldn't tell anything was checked out or locked from the UI.
